web/src/emoji_picker.js

```diff
@@ -619,79 +620,66 @@ function register_popover_events($popover) {
     });
 }
 
-export function build_emoji_popover($elt, id) {
-    const template_args = {
-        class: "emoji-info-popover",
-    };
-    let placement = popovers.compute_placement($elt, APPROX_HEIGHT, APPROX_WIDTH, true);
-
-    if (placement === "viewport_center") {
-        // For legacy reasons `compute_placement` actually can
-        // return `viewport_center`, but bootstrap doesn't actually
-        // support that.
-        placement = "left";
-    }
-
-    let template = render_emoji_popover(template_args);
-
-    // if the window is mobile sized, add the `.popover-flex` wrapper to the emoji
-    // popover so that it will be wrapped in flex and centered in the screen.
-    if (window.innerWidth <= 768) {
-        template = "<div class='popover-flex'>" + template + "</div>";
-    }
-
-    $elt.popover({
-        // temporary patch for handling popover placement of `viewport_center`
-        placement,
-        fix_positions: true,
-        template,
-        title: "",
-        content: generate_emoji_picker_content(id),
-        html: true,
-        trigger: "manual",
-        fixed: true,
-    });
-    $elt.popover("show");
-
-    const $popover = $elt.data("popover").$tip;
-    $popover.find(".emoji-popover-filter").trigger("focus");
-    $current_message_emoji_popover_elem = $elt;
-
-    emoji_catalog_last_coordinates = {
-        section: 0,
-        index: 0,
+function get_default_emoji_popover_options(template) {
+    return {
+        placement: "top",
+        onCreate(instance) {
+            if (current_message_id) {
+                message_lists.current.select_id(current_message_id);
+            }
+            emoji_popover_instance = instance;
+            const $popover = $(instance.popper);
+            $popover.addClass("emoji-popover-root");
+            instance.setContent(ui_util.parse_html(template));
```

Comment:
Do we need to be using this function? Can you explain why it's necessary and safe? 

```
/**                                                                                                     
 * Parse HTML and return a DocumentFragment.                                                            
 *                                                                                                      
 * Like any consumer of HTML, this function must only be given input                                    
 * from trusted producers of safe HTML, such as auto-escaping                                           
 * templates; violating this expectation will introduce bugs that are                                   
 * likely to be security vulnerabilities.                                                               
 */                                                                                                     
export function parse_html(html: string): DocumentFragment {                                            
    const template = document.createElement("template");                                                
    template.innerHTML = html;                                                                          
    return template.content;                                                                            
}                                                                                                       
```

Reply:
I guess I'd feel a lot more comfortable if you just had `render_emoji_popover()` inline here, which seems like the only value `template` ever has? It's a very weird design to have that parameter.

Reply:
The `setContent` method expects an HTML element, not a string in this case. That's why we use the `parse_html` function around the codebase.

Reply:
> I guess I'd feel a lot more comfortable if you just had render_emoji_popover() inline here, which seems like the only value template ever has? It's a very weird design to have that parameter.

Yeah, I agree. I changed it to an inline value.